Source/WebKit/UIProcess/AuxiliaryProcessProxy.cpp

```diff
@@ -335,6 +335,8 @@ void AuxiliaryProcessProxy::didFinishLaunching(ProcessLauncher* launcher, IPC::C
     if (!connectionIdentifier)
         return;
 
+    m_initializationActivity = throttler().foregroundActivity("Process initialization Activity"_s).moveToUniquePtr();
```

Comment:
Why take a foreground activity in didFinishLaunching, why not take it when starting to launch the process? Our ProcessAssertion logic will take care of taking the right assertion as soon as we get a PID already. There is no reason to delay the taking of the activity until we have a PID.

Reply:
Good catch! I'll move this to where we start launching the process.